### ISSUE TITLE
Custom statements don't support ids (breaks autovoice, extract_dialog, more?)

### ISSUE THREAD

**Issue by cisco-donovan** [renpy/renpy#4290]
Hello, I've hit some trouble lately with custom statements. I can't see any discussion or documentation around this so I figure it's worth raising an issue.

In my game, I've overridden the default say statement with `renpy.register_statement("")`. Turns out this breaks linting stats, extract_dialogue and auto voice because custom say statements don't generate or support ids. Translation also gets harder to maintain, imo, if I can't control the ids of every line.

In the short term I'd recommend adding some notes to the `register_statement` docs about this.

**Comment by Gouvernathor** [comment]
I don't think this is something relative to register_statement and creator-defined statements, but rather something to do with the say statement and the limitation of its python equivalent(s).

Your statement presumably parses all the parameters manually, and passes them to the python equivalents of the renpy statements whose behavior you want to copy. But then, you're bound by the limitations of these statement equivalents. In say's case, the id and the translation system are not covered by renpy.say or the other python equivalents, and never have been.

The launcher not being able to generate your translations is normal : it looks for our say statement, and there are none. There is only your statement which is not a say statement, only something *behaving* the same way. Custom say statements would be able to support an id system if you code it in, but then it wouldn't be the same id system as renpy's native one, and you would need to create a translate block yourself (as another CDS) and write each translation yourself.

This could be considered a feature request into supporting a statement equivalent for the translation system. But I don't see it as a flaw in the CDS system.

**Comment by cisco-donovan** [comment]
@Gouvernathor Thank you for the response.

I think the priority here is to make it a bit clearer in the docs what the consquences overridding the default statement are. It's not something that most people will try and do, but for idiots like me a little warning would be appreciated.

I'm currently looking into workarounds so that I can use the default say statemenet, which I'm sure will be fine!

**Comment by Gouvernathor** [comment]
I think I agree, in the current state of things it's probably a bit foolhardy to do that without deep prior knowledge of how CDS work, and we should probably warn against that.

**Comment by Andykl** [comment]
I believe this is resolved.
